src/com/xilinx/rapidwright/design/ConstraintsTools.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2025, Advanced Micro Devices, Inc.
+ * All rights reserved.
+ *
+ * Author: Coherent Ho, Synopsys, Inc.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.design;
+
+import java.util.HashMap;
+import com.xilinx.rapidwright.design.blocks.PBlock;
+import com.xilinx.rapidwright.design.ConstraintGroup;
+
+/**
+ * A collection of methods to access design constraints.
+ *
+ * Created on: Oct 31, 2025
+ */
+public class ConstraintsTools {
+    private static HashMap<String, PBlock> pblockMap = new HashMap<>();
+    private Design d;
+
+    public ConstraintsTools(Design d){
+        this.d = d;
+        parsePBlockFromXDCConstraints();
+    }
+
+    private void parsePBlockFromXDCConstraints(){
```

Comment:
Sorry, a few more comments.  Could we change this method so that it is `static` and takes as input a `Design` object and returns the `Map<String, PBlock>`?  That would follow the pattern of the `*Tool` classes.  We typically don't want to instantiate classes like this that are a collection of tools.  To be clear, this would be the new signature:
```
public static Map<String, PBlock> parsePBlocksFromXDC(Design design) 
```
Then you could remove the constructors and other methods.